### ISSUE TITLE
redundantSelf doesn't autocorrect when using `if case` inside a struct

### ISSUE THREAD

**Issue by marcelofabri** [nicklockwood/SwiftFormat#1409]
Given this file:

```swift
struct A {
    func doSomething() {
        B.method { mode in
            if case .edit = mode {
                self.doA()
            } else {
                self.doB()
            }
        }
    }

    func doA() {}
    func doB() {}
}
```

I only get one warning/correction in the second `self.`:

```
$ swiftformat --lint file.swift
Running SwiftFormat...
(lint mode - no files will be changed.)
Reading config file at /Users/marcelo.fabri/my-secret-project/.swiftformat
Reading swift-version file at /Users/marcelo.fabri/my-secret-project/.swift-version (version 5.4)
/Users/marcelo.fabri/my-secret-project/file.swift:7:1: warning: (redundantSelf) Insert/remove explicit self where applicable.
SwiftFormat completed in 0.66s.
Source input did not pass lint check.
1/1 files require formatting.
```

This is using 0.51.4. If I change `if case` to a regular `if`, everything works as expected.

**Comment by nicklockwood** [comment]
@marcelofabri wait, isn't it unsafe to remove `self` in this case anyway since it's inside a closure?

**Comment by marcelofabri** [comment]
> @marcelofabri wait, isn't it unsafe to remove `self` in this case anyway since it's inside a closure?
> 
> EDIT: ah wait, never mind, this is using swift version 5.8, right?

We are not using 5.8, but it should be safe as it’s a struct - or at least the compiler was ok with this change, idk if there’s a case where it’d break.

**Comment by nicklockwood** [comment]
@marcelofabri fixed in 0.51.5
